### PR TITLE
Global Collect Success Criteria

### PR DESCRIPTION
**Pull request by peteroas**
Provides more action-specific success Criteria for Global Collect's payment gateway, with accompanying test coverage.

### CODE REVIEW

File: lib/active_merchant/billing/gateways/global_collect.rb [hunk 1]
```diff
@@ -457,8 +457,7 @@ def success_from(action, response)
           refund_status = response.dig('status') || response.dig('payment', 'status')
           %w(REFUNDED REFUND_REQUESTED).include?(refund_status)
         else
-          response['status'] != 'REJECTED' # default prior to revision, should be removed after building out the rest
-        end
```
Comment (aenand) [inline lib/active_merchant/billing/gateways/global_collect.rb]: Did you mean to remove this `end` statement?

File: lib/active_merchant/billing/gateways/global_collect.rb [hunk 2]
```diff
@@ -458,13 +475,7 @@ def message_from(succeeded, response)
       end
 
       def authorization_from(succeeded, response)
```
Comment (aenand) [inline lib/active_merchant/billing/gateways/global_collect.rb:477]: Do we still need succeeded as a parameter here? It doesn't seem like we use it in this method

**Review by aenand — CHANGES_REQUESTED** [review]
Looks good. I left a couple points of feedback, the removal of the `end` line might be causing the ruby checks to fail?

File: lib/active_merchant/billing/gateways/global_collect.rb (hunk 1 above)
Comment (peteroas) [reply lib/active_merchant/billing/gateways/global_collect.rb]: lol that was the issue hah

**Review by aenand — COMMENTED** [review]
I have 3 remote tests failing. All three have the same message: "AUTHORISATION FAILED FOR REQUEST {{request_id}}"
`test_successful_purchase_with_naranja`
`test_successful_purchase_with_installments`
`test_successful_purchase_with_cabal`

**Comment by curiousepic** [comment]
@aenand FYI those are unrelated to this change and fail on master as well

File: lib/active_merchant/billing/gateways/global_collect.rb (hunk 2 above)
Comment (peteroas) [reply lib/active_merchant/billing/gateways/global_collect.rb:478]: good point, will revise this!

**Review by aenand — APPROVED** [review]
Code looks good and tests (except the 3 mentioned) pass. 🚀

File: test/remote/gateways/remote_global_collect_test.rb [hunk 3]
```diff
@@ -267,12 +267,28 @@ def test_successful_purchase_with_blank_name
...
 
+<<<<<<< HEAD
+=======
+  def test_successful_purchase_with_pre_authorization_flag
+    response = @gateway.purchase(@accepted_amount, @credit_card, @options.merge(pre_authorization: true))
+    assert_success response
+    assert_equal 'Succeeded', response.message
+  end
+
+>>>>>>> 9605cc86 (fixes rubocop linting)
```
Comment (curiousepic) [inline test/remote/gateways/remote_global_collect_test.rb:278]: clean up conflict notation

File: CHANGELOG [hunk 4]
```diff
@@ -2,6 +2,7 @@
 = ActiveMerchant CHANGELOG
 
 == HEAD
+* GlobalCollect: Provides more action-specific success Criteria for Global Collect's payment gatewa [poas14] #4308
```
Comment (curiousepic) [inline CHANGELOG:5]: New entries go at the bottom of the HEAD list. Also, we usually wait to add these until the merge so we don't have ordering conflicts.

Comment (peteroas) [reply CHANGELOG:5]: ahh gotcha, so you just add it immediately before merging? will delete for now until merge

**Review by curiousepic — APPROVED** [review]
Looks good!  Thanks for picking this up where I left off. Just be sure to squash to one commit before merging. 👍

**Pull request closed without merging**